API Display Customizations

softworkz opened the pull request:
- Improve display of response schemas and display those by default
- Add filtering support
- Don't expand by default
- Added static view that does not require an emby server (https://softworkz.github.io/swagger-ui/?staticview=true). This could be added as a link to the documentation Wiki and allows browsing the API without having an active server. This view does not have Try-Out and authorize buttons

LukePulverenti commented:
Can i just take the index.html changes? I want to be able to easily merge new swagger releases. Currently index.html is the only file that is customized. It looks like this will make it harder to update to new releases.

softworkz commented:
Yea, I thought about the same after submitting. I just made about 3 or 4 changes to the css, that's all.

Hold on, I'll wrap those changes into a separate css to override the existing css classes, then we can leave the original css untouched.

softworkz committed:
Ignore web.config

softworkz committed:
Customizations for Swagger-UI

softworkz committed:
Add latest API descrription

softworkz commented:
I've changed this now in a way that the original files are left untouched.

Also I added some improvements for rendering schema definitions. It's a much more streamlined view with less distraction, more information per area and signiificantly reduced number of expander arrows to click on.

FYI: I will not charge for all the beautification aspects of this.
I've actually done it out of personal interest to learn a bit about what React is and how if works.
But I've also been pretty annoyed by their ugly schema presentation and wanted to have something better for us :-)
